commons/src/main/java/info/metadude/android/eventfahrplan/commons/temporal/DateFormatter.kt

```diff
@@ -39,15 +41,29 @@ class DateFormatter private constructor() {
     /**
      * Returns a formatted string suitable for sharing it with people worldwide.
      * It contains day, month, year and time in current system locale in long format
-     * and the associated time zone offset.
-     * E.g. Tuesday, January 22, 2019 1:00 AM GMT+01:00
+     * and the associated time zone offset. The [timeZoneId] is appended if it is valid.
+     * E.g. Tuesday, January 22, 2019 1:00 AM GMT+01:00 (Europe/Berlin)
+     *
+     * The [timeZoneId] is expected to be given as a region-based zone ID which has the form
+     * '{area}/{city}' such as 'Europe/Paris' or 'America/New_York'.
      */
-    fun getFormattedShareable(time: Long): String {
-        // TODO: Use time zone of the event rather than the device time zone.
-        val displayTimeZone = ZoneId.systemDefault()
+    fun getFormattedShareable(time: Long, timeZoneId: String): String {
+        var isTimeZoneIdValid = false
+        var displayTimeZone: ZoneId
+        try {
+            displayTimeZone = ZoneId.of(timeZoneId)
+            isTimeZoneIdValid = true
+        } catch (e: DateTimeException) {
+            displayTimeZone = ZoneId.systemDefault()
+        } catch (e: ZoneRulesException) {
+            displayTimeZone = ZoneId.systemDefault()
+        }
         val timeZoneOffset = timeZoneOffsetFormatter.withZone(displayTimeZone).format(Instant.ofEpochMilli(time))
-        // TODO Append time zone name once it is provided. See https://github.com/EventFahrplan/EventFahrplan/pull/296.
-        return "${dateTimeFull.format(Date(time))} $timeZoneOffset"
+        var shareableText = "${dateTimeFull.format(Date(time))} $timeZoneOffset"
```

Comment:
I was afraid this might happen given the code structure. Now `timeZoneOffset` might use the event's time zone information but `dateTimeFull` is not explicitly configured with a time zone so it's using the time zone of the device.

Reply:
Following up at our offline talk I updated the implementation with the following main points:

- `DateFormatter` is based on threetenbp library now. System time zone is used.
- `Meta#timeZoneName` is nullable; database column is `null` by default.
- Validation is applied to time zone name before writing its value to the database.
- `ZoneId?` is used in layers above the database layer for passing the parsed time zone name as `timeZoneId`.
- `dateFullTimeShortFormatter` in `DateFormatter` takes the time zone of the event into account with a fallback to the system default.